### PR TITLE
feat: Add messaging for free seats and clarify paid

### PR DESCRIPTION
**Pull request by calvin-codecov**
Closes https://linear.app/getsentry/issue/CCMRG-1333/billing-ui-should-display-free-seats-as-a-clear-part-of-total-seats
Relies on https://github.com/codecov/umbrella/pull/413 for GQL info

This PR is employing a couple of techniques to better clarify paid vs free seats. Since free seats are uncommon, we generally do not want to add too much extra text for all use cases as it will be irrelevant for most.

- show number of free seats only if they have any free seats
- specify "paid" seats for general wording

Side note: any of the benefits list bullet point text for individual plans come from DB entries which have been updated, instead of gazebo code.

NOTE: Most of the changes are just updating test data to include `freeSeatCount` which now comes from GQL.

Screenshot 2025-09-02 at 4 15 08 PM

### CODE REVIEW

File: src/shared/utils/billing.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export const formatNumberToUSD = (value: number) =>
 export function getNextBillingDate(
   accountDetails?: z.infer<typeof AccountDetailsSchema> | null
 ) {
-  const timestamp = accountDetails?.subscriptionDetail?.latestInvoice?.periodEnd
+  const timestamp = accountDetails?.subscriptionDetail?.currentPeriodEnd
```
Comment (calvin-codecov) [inline src/shared/utils/billing.ts:134]: This change makes this consistent with the uses in PaymentCard and DowngradePlan. `currentPeriodEnd` looked buggy - in test, I started a new subscription and `currentPeriodStart` and `currentPeriodEnd` was the same value coming back from stripe

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/PaymentCard/PaymentCard.jsx [hunk 2]
```diff
@@ -2,26 +2,62 @@ import PropTypes from 'prop-types'
 import { useState } from 'react'
 
 import { accountDetailsPropType } from 'services/account/propTypes'
-import { formatTimestampToCalendarDate } from 'shared/utils/billing'
+import { usePlanData } from 'services/account/usePlanData'
```
Comment (calvin-codecov) [inline src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/PaymentCard/PaymentCard.jsx:5]: this file's changes are new content

File: src/pages/MembersPage/MembersActivation/Activation/Activation.jsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import pluralize from 'pluralize'
```
Comment (ajay-sentry) [inline src/pages/MembersPage/MembersActivation/Activation/Activation.jsx:1]: is pluralize imported by default? I only see the types in the package.json so just wanna be sure

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/PaymentCard/PaymentCard.jsx [hunk 4]
```diff
@@ -2,26 +2,62 @@ import PropTypes from 'prop-types'
...
 function PaymentCard({ accountDetails, provider, owner }) {
   const [isFormOpen, setIsFormOpen] = useState(false)
+  const { data: planData } = usePlanData({
+    provider,
+    owner,
+  })
   const subscriptionDetail = accountDetails?.subscriptionDetail
   const card = subscriptionDetail?.defaultPaymentMethod?.card
   const usBankAccount = subscriptionDetail?.defaultPaymentMethod?.usBankAccount
+  const isPerYear = planData?.plan?.billingRate === BillingRate.ANNUALLY
```
Comment (ajay-sentry) [inline src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/PaymentCard/PaymentCard.jsx:33]: should we put this stuff in a useCallback or smth?

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/PaymentCard/PaymentCard.jsx [hunk 5]
```diff
@@ -2,26 +2,62 @@ import PropTypes from 'prop-types'
...
+        })
 
   let nextBillingDisplayDate = null
   if (!subscriptionDetail?.cancelAtPeriodEnd) {
     nextBillingDisplayDate = formatTimestampToCalendarDate(
       subscriptionDetail?.currentPeriodEnd
     )
   }
+  const nextBillPrice = formatNumberToUSD(
+    isPerYear ? billPrice * 12 : billPrice
```
Comment (ajay-sentry) [inline src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/PaymentCard/PaymentCard.jsx:59]: nit: magic number

File: src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/PlanTypeOptions/PlanTypeOptions.tsx [hunk 6]
```diff
@@ -110,7 +110,10 @@ const PlanTypeOptions: React.FC<PlanTypeOptionsProps> = ({
             </RadioTileGroup.Item>
           </RadioTileGroup>
           {planOption === TierName.TEAM && (
-            <p>Up to {TEAM_PLAN_MAX_ACTIVE_USERS} users</p>
+            <p>
+              Up to {TEAM_PLAN_MAX_ACTIVE_USERS}
+              {planData?.plan?.freeSeatCount ? ' paid' : ''} users
```
Comment (ajay-sentry) [inline src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/PlanTypeOptions/PlanTypeOptions.tsx:115]: I think we could probably just say "paid" here regardless of the freeSeatCount, right?

File: src/pages/MembersPage/MembersActivation/Activation/Activation.jsx (hunk 3 above)
Comment (calvin-codecov) [reply src/pages/MembersPage/MembersActivation/Activation/Activation.jsx:1]: yup, it's already in the package.json

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/PaidPlanCard/PaidPlanCard.tsx [hunk 7]
```diff
@@ -64,6 +71,9 @@ function PaidPlanCard() {
             {seats ? (
               <p className="text-xs text-ds-gray-senary">
                 plan has {seats} seats
+                {freeSeats && seats - freeSeats > 0
```
[the remark this answers is not in the document]
Comment (calvin-codecov) [reply src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/PaidPlanCard/PaidPlanCard.tsx:74]: Are you suggesting add parens around the copy rendered or parens around part of the code?

File: src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/PlanTypeOptions/PlanTypeOptions.tsx (hunk 6 above)
Comment (calvin-codecov) [reply src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/PlanTypeOptions/PlanTypeOptions.tsx:115]: Adalene and I were thinking of leaving out "paid" as often as possible for simplification when a user doesn't have free seats because it should be a very small percentage of total users. I could go either way this though since we decided on this before I realized the plan card benefits list can't dynamically decide on adding "paid" since they're from the DB so one could argue we just say "paid" everywhere for consistency. Thoughts?

File: src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/PaidPlanCard/PaidPlanCard.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ajay-sentry) [reply src/pages/PlanPage/subRoutes/CurrentOrgPlan/CurrentPlanCard/PaidPlanCard/PaidPlanCard.tsx:74]: parens around ((seats - freeSeats) > 0) or similar

File: src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/PlanTypeOptions/PlanTypeOptions.tsx (hunk 6 above)
Comment (ajay-sentry) [reply src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/PlanTypeOptions/PlanTypeOptions.tsx:115]: I'm on team paid everywhere personally but I can see why we decided to do the conditional. I think I saw somewhere else in the code where we added paid unconditionally though (in this PR)

File: src/shared/utils/upgradeForm.test.ts [hunk 8]
```diff
@@ -214,6 +234,56 @@ describe('getDefaultValuesUpgradeForm', () => {
       seats: 2,
     })
   })
+
+  describe('quantity calculation edge cases', () => {
+    it('returns 0 when planUserCount is null', () => {
```
Comment (ajay-sentry) [inline src/shared/utils/upgradeForm.test.ts:239]: this guy
